datahub-web-react/src/app/search/highlight/SearchTextHighlighter.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import Highlight from 'react-highlighter';
+import styled from 'styled-components';
+import { useMatchedFieldsByGroup } from '../context/SearchResultContext';
+import { useSearchQuery } from '../context/SearchContext';
+import { MatchedFieldName } from '../context/constants';
+
+type Props = {
+    field: MatchedFieldName;
+    text: string;
+    enableFullHighlight?: boolean;
+};
+
+const HIGHLIGHT_ALL_PATTERN = /.*/;
+
+const StyledHighlight = styled(Highlight).attrs((props) => ({
+    matchStyle: { background: props.theme.styles['highlight-color'] },
+}))``;
+
+const SearchTextHighlighter = ({ field, text, enableFullHighlight = false }: Props) => {
+    const matchedFields = useMatchedFieldsByGroup(field);
+    const hasMatchedField = !!matchedFields?.length;
+    const normalizedSearchQuery = useSearchQuery()?.trim().toLowerCase();
+    const normalizedText = text.trim().toLowerCase();
+    const hasSubstring = hasMatchedField && !!normalizedSearchQuery && normalizedText.includes(normalizedSearchQuery);
+    const pattern = enableFullHighlight ? HIGHLIGHT_ALL_PATTERN : undefined;
```

Comment:
i'm a little confused by the full highlight situation - why do we want to highlight the full text if it doesn't match anything in the search query? after pulling down I'm seeing situations like this where my search query is `long_tail` and `ADOPTIONS` is being highlighted (but I believe these results are showing up because `long_tail` matches their container's name)
<img width="1716" alt="image" src="https://github.com/datahub-project/datahub/assets/28656603/f7dcacf8-6ead-4bc7-b274-a6394abfd95f">

Reply:
Removed qualifiedName since we only use that for container names and the solve would be complex.